jdaviz/configs/mosviz/plugins/parsers.py

```diff
@@ -738,7 +739,10 @@ def mos_niriss_parser(app, data_dir, instrument=None,
         for image_file in files_by_labels["Direct Image"]:
             # save label for table viewer
             im_split = image_file.stem.split("_")[0]
-            pupil = fits.getheader(image_file, ext=0).get('PUPIL')
+            if instrument == "niriss":
+                pupil = fits.getheader(image_file, ext=0).get('PUPIL')
+            elif instrument == "nircam":
+                pupil = fits.getheader(image_file, ext=0).get('FILTER')
```

Comment:
Should we add an `else` here or by the time you get here, it is either niriss or nircam and nothing else?

Reply:
Right now those are the only two possible values, `nirspec` uses a different parser and anything else throws an error in `load_data`.

Reply:
In that case, maybe less confusing to change

```python
elif instrument == "nircam":
```

to this?

```python
else:  # nircam
```

Reply:
I'm inclined to leave it as-is, since if we unify another instrument into this parser and forget to add a case for it, it will throw an error, as opposed to maybe being handled incorrectly by the `else` statement.

Reply:
Then isn't it slightly prettier like this:

```python
else:  # pragma: no cover
    raise NotImplementedError(f'{instrument} is not supported')
```